### PR TITLE
DEVDOCS-3840: [update] Add channel_id as query parameter

### PR DESCRIPTION
**Pull request by bc-traciporter**
# [DEVDOCS-3840]

## What changed?
Added channel_id as a query parameter

## Anything else?

[DEVDOCS-3840]: https://bigcommercecloud.atlassian.net/browse/DEVDOCS-3840?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reference/catalog.v3.yml [hunk 1]
```diff
@@ -409,6 +409,16 @@ paths:
             type: array
             items:
               type: string
+        - schema:
+            type: string
+          in: query
+          name: channel_id
```
Comment (kristinapototska) [inline reference/catalog.v3.yml:415]: Can you please check the description of this ticket and the implementation https://bigcommercecloud.atlassian.net/browse/MSF-2510 
as it was updated to be an array of `channels`

**Review by kristinapototska — CHANGES_REQUESTED** [review]

File: reference/catalog.v3.yml [hunk 2]
```diff
@@ -903,8 +1158,8 @@ paths:
         - 250 characters product name length.
 
         **Usage Notes**
-        * `POST` requests to `/products` accept a `video` array. 
-        * `POST` requests to `/products/{product_id}/videos` accept a `video` object.
+        * `POST` requests to `/products` accepts a single `video` object; to send an array of video objects, see: `/products/{product_id}/videos`.
```
Comment (slsriehl) [inline reference/catalog.v3.yml:1161]:
```suggestion
        * `POST` requests to `/products` accept a single `video` object. To send an array of video objects, use `/products/{product_id}/videos`.
```

File: reference/catalog.v3.yml [hunk 3]
```diff
@@ -903,8 +1158,8 @@ paths:
...
+        * You can supply category IDs, but the Channel ID array should remain empty.
```
Comment (slsriehl) [inline reference/catalog.v3.yml:1162]:
```suggestion
        * You can supply category IDs, but the channel ID array should remain empty.
```

File: reference/catalog.v3.yml [hunk 4]
```diff
@@ -439,6 +691,9 @@ paths:
         - `date_modified`
         - `calculated_price`
         - `base_variant_id`
+
+        **Usage Notes**
+        You can supply category IDs, but the Channel ID array should remain empty.
```
Comment (slsriehl) [inline reference/catalog.v3.yml:696]:
```suggestion
        You can supply category IDs, but the channel ID array should remain empty.
```

File: reference/catalog.v3.yml [hunk 5]
```diff
@@ -1556,6 +1916,9 @@ paths:
         - date_modified
         - calculated_price
         - base_variant_id
+
+        **Usage Notes**
+        * You can supply category IDs, but the Channel ID array should remain empty.
```
Comment (slsriehl) [inline reference/catalog.v3.yml:1921]:
```suggestion
        * You can supply category IDs, but the channel ID array should remain empty.
```

File: reference/catalog.v3.yml [hunk 6]
```diff
@@ -24148,6 +24514,12 @@ components:
           description: |
             The total quantity of this product sold.
           example: 80
+        channels:
+          type: array
+          description: 'An optional field that can be empty, or you can supply a single ID or an array of IDs.'
```
Comment (slsriehl) [inline reference/catalog.v3.yml:24519]: are we 100% that you can actually pass a single ID instead of an array? this seems unlikely.

Comment (slsriehl) [reply reference/catalog.v3.yml:24519]: if it is true, then the datatype is a oneOf situation.

File: reference/catalog.v3.yml [hunk 7]
```diff
@@ -87,7 +87,9 @@ paths:
       tags:
         - Products
       summary: Get All Products
-      description: Returns a list of **Products**. Optional filter parameters can be passed in.
+      description: |-
+        Returns a list of **Products**. Optional filter parameters can be passed in.
+        You can supply category IDs, but the Channel ID array should remain empty.
```
Comment (slsriehl) [inline reference/catalog.v3.yml:92]:
```suggestion
        You can supply category IDs, but the channel ID array should remain empty.
```

File: reference/catalog.v3.yml (hunk 1 above)
Comment (slsriehl) [reply reference/catalog.v3.yml:415]: @kristinapototska please re-review

File: reference/catalog.v3.yml (hunk 6 above)
Comment (bc-traciporter) [reply reference/catalog.v3.yml:24519]: @slsriehl Kristina explained that it is always an array. The array can have a single ID or multiple IDs in the array.

**Review by kristinapototska — CHANGES_REQUESTED** [review]

File: reference/catalog.v3.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (bc-traciporter) [reply reference/catalog.v3.yml:92]: @kristinapototska when testing the Get All Products endpoint, I used `channels`: [1], and it worked.

```
{
  "id": 107,
  "name": "Dustpan & Brush",
  "type": "physical",
  "sku": "DPB",
  "description": "<p><span>Lorem ipsum dolor sit amet, consectetur adipiscing elit. Morbi vel metus ac est egestas porta sed quis erat. Integer id nulla massa. Proin vitae enim nisi. Praesent non dignissim nulla. Nulla mattis id massa ac pharetra. Mauris et nisi in dolor aliquam sodales. Aliquam dui nisl, dictum quis leo sit amet, rutrum volutpat metus. Curabitur libero nunc, interdum ac libero non, tristique porttitor metus. Ut non dignissim lorem, in vestibulum leo. Vivamus sodales quis turpis eget.</span></p>",
  "weight": 5,
  "width": 10,
  "depth": 5,
  "height": 5,
  "price": 20,
  "cost_price": 20,
  "retail_price": 20,
  "sale_price": 20,
  "map_price": 20,
  "tax_class_id": 0,
  "product_tax_code": "string",
  "categories": [
    23,
    22
  ],
  "channels": [
      1
  ],
  "brand_id": 36,
  "inventory_level": 0,
  "inventory_warning_level": 0,
  "inventory_tracking": "none",
  "fixed_cost_shipping_price": 0,
  "is_free_shipping": true,
  "is_visible": true,
  "is_featured": true,
  "related_products": [
    0
  ],
  "warranty": "string",
  "bin_picking_number": "string",
  "layout_file": "string",
  "upc": "string",
  "search_keywords": "string",
  "availability_description": "string",
  "availability": "available",
  "gift_wrapping_options_type": "any",
  "gift_wrapping_options_list": [
  ],
  "sort_order": -2147483648,
  "condition": "New",
  "is_condition_shown": true,
  "order_quantity_minimum": 0,
  "order_quantity_maximum": 0,
  "page_title": "string",
  "meta_keywords": [
    "string"
  ],
  "meta_description": "string",
  "view_count": 0,
  "preorder_release_date": "2019-08-24T14:15:22Z",
  "preorder_message": "string",
  "is_preorder_only": true,
  "is_price_hidden": true,
  "price_hidden_label": "string",
  "custom_url": {
    "url": "string",
    "is_customized": true
  },
  "open_graph_type": "product",
  "open_graph_title": "string",
  "open_graph_description": "string",
  "open_graph_use_meta_description": true,
  "open_graph_use_product_name": true,
  "open_graph_use_image": true,
  "brand_name or brand_id": "Common Good",
  "gtin": "string",
  "mpn": "string",
  "reviews_rating_sum": 3,
  "reviews_count": 4,
  "total_sold": 80,
  "images": [
    {
      "image_file": "string",
      "is_thumbnail": true,
      "sort_order": -2147483648,
      "description": "string",
      "image_url": "https://cdn11.bigcommerce.com/s-w0astee9jo/products/126/images/386/palestine48hoodie-350x350__58794.1641415487.220.290.jpg?c=1",
      "id": 0,
      "product_id": 0,
      "url_zoom": "string",
      "url_standard": "string",
      "url_thumbnail": "string",
      "url_tiny": "string",
      "date_modified": "2019-08-24T14:15:22Z"
    }
  ],
  "videos": [
    {
      "title": "Writing Great Documentation",
      "description": "A video about documenation",
      "sort_order": 1,
      "type": "youtube",
      "video_id": "z3fRu9pkuXE",
      "id": 0,
      "product_id": 0,
      "length": "string"
    }
  ]
}
]
```

File: reference/catalog.v3.yml [hunk 8]
```diff
@@ -1555,6 +1916,9 @@ paths:
         - date_modified
         - calculated_price
         - base_variant_id
+
+        **Usage Notes**
+        * You can supply category IDs, but the Channel ID array should remain empty.
```
[the remark this answers is not in the document]
Comment (bc-traciporter) [reply reference/catalog.v3.yml:1921]: @kristinapototska I think it should only say "You can supply category IDs."

File: reference/catalog.v3.yml (hunk 7 above)
[the remark this answers is not in the document]
Comment (bc-traciporter) [reply reference/catalog.v3.yml:92]: I want to change it to only say " You can supply category IDs."

File: reference/catalog.v3.yml [hunk 9]
```diff
@@ -24148,6 +24514,12 @@ components:
           description: |
             The total quantity of this product sold.
           example: 80
+        channels:
+          type: array
+          description: 'An optional field that can be empty, or you can supply a single ID or multiple IDs.'
```
Comment (slsriehl) [inline reference/catalog.v3.yml:24519]:
```suggestion
          description: 'Optional. You can supply a single ID, multiple IDs, or leave the field empty.'
```

File: reference/catalog.v3.yml [hunk 10]
```diff
@@ -439,6 +691,9 @@ paths:
         - `date_modified`
         - `calculated_price`
         - `base_variant_id`
+
+        **Usage Notes**
+        You can supply category IDs, but the channel ID array should remain empty.
```
Comment (slsriehl) [inline reference/catalog.v3.yml:696]: elsewhere we've changed this. do we want to keep the `channel ID array should remain empty` in this location?

File: reference/catalog.v3.yml [hunk 11]
```diff
@@ -903,8 +1158,8 @@ paths:
         - 250 characters product name length.
 
         **Usage Notes**
-        * `POST` requests to `/products` accept a `video` array. 
-        * `POST` requests to `/products/{product_id}/videos` accept a `video` object.
+        * `POST` requests to `/products` accept a single `video` object. To send an array of video objects, use: `/products/{product_id}/videos`.
+        * You can supply category IDs, but the channel ID array should remain empty.
```
Comment (slsriehl) [inline reference/catalog.v3.yml:1162]: elsewhere we've changed this. do we want to keep the `channel ID array should remain empty` in this location?

**Review by slsriehl — APPROVED** [review]
👍🏻

**Pull request merged**